### PR TITLE
Add detailed balance info to WalletBalance RPC

### LINKED ISSUE #415: Add unconfirmed wallet balance to walletbalance RPC
Currently, the walletbalance RPC returns the fully confirmed wallet balance. This can be confusing when a channel is opened with a particular amount, but a larger amount appears to be subtracted from the wallet balance until the change transaction is confirmed. This also creates confusion in the App UI. One solution would be to have the RPC return both the balance and the unconfirmed amount, and have the App display both. (Originally reported by several users via Slack, including moli and fei.)

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Replaces current `balance` field by 3 fields, `TotalBalance`, `ConfirmedBalance` and `UnconfirmedBalance` to store the respective balance information.

Closes #415.

### CODE REVIEW

File: rpcserver.go [hunk 1]
```diff
@@ -1079,15 +1078,27 @@ func (r *rpcServer) WalletBalance(ctx context.Context,
 		}
 	}
 
-	balance, err := r.server.cc.wallet.ConfirmedBalance(1, in.WitnessOnly)
+	// Get total balance, from txs that have >= 0 confirmations
+	TotalBal, err := r.server.cc.wallet.ConfirmedBalance(0, in.WitnessOnly)
```
Comment (mlerner) [inline rpcserver.go]: nit: make `TotalBal` => `totalBal` because this is an intermediate variable.

File: rpcserver.go [hunk 2]
```diff
@@ -1079,15 +1078,27 @@ func (r *rpcServer) WalletBalance(ctx context.Context,
...
 	if err != nil {
 		return nil, err
 	}
 
-	rpcsLog.Debugf("[walletbalance] balance=%v", balance)
+	// Get confirmed balance, from txs that have >= 1 confirmations
+	ConfirmedBal, err := r.server.cc.wallet.ConfirmedBalance(1, in.WitnessOnly)
```
Comment (mlerner) [inline rpcserver.go]: nit: make `ConfirmedBal ` => `confirmedBal` because this is an intermediate variable.

File: rpcserver.go [hunk 3]
```diff
@@ -1063,11 +1063,10 @@ func (r *rpcServer) ListPeers(ctx context.Context,
 	return resp, nil
 }
 
-// WalletBalance returns the sum of all confirmed unspent outputs under control
+// WalletBalance returns total unspent outputs(confirmed or unconfirmed), all
+// confirmed unspent outputs and all unconfirmed unspent outputs under control
 // by the wallet. This method can be modified by having the request specify
 // only witness outputs should be factored into the final output sum.
-// TODO(roasbeef): split into total and confirmed/unconfirmed
-// TODO(roasbeef): add async hooks into wallet balance changes
```
Comment (mlerner) [inline rpcserver.go]: I think this TODO needs to be added back - it is not implemented or in scope for this PR.

**Comment by mlerner** [comment]
Two other things: you will need to rebase your branch with master to resolve, and also reformat commit messages, as mentioned in the style guide -
 https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#44-model-git-commit-messages

All commits should begin with the subsystem or package primarily affected by the change - in this case you could split up your commits into `proto` (for the `*.proto` changes) and `lncli` for code that uses those changes.

**Comment by nalinbhardwaj** [comment]
@mlerner : I think I've addressed all changes you mentioned now. Thanks for letting me know. I'll do similar changes with #417 commit messages as well. :)

Note: Seems like the travis build is failing for unrelated reasons.

**Comment by nalinbhardwaj** [comment]
@Roasbeef : Addressed the nits, sorry about those.

**Comment by Roasbeef** [comment]
Hey @nalinbhardwaj, can you rebase this? A recent PR went in that also modified the protos. You'll just need to rebuild them.

**Comment by nalinbhardwaj** [comment]
@roasbeef done. Not sure why there’s a commit you made “with me” though? I don’t see a way to remove that offline, let me know if that’s a problem.

**Comment by Roasbeef** [comment]
@nalinbhardwaj you'll need to remove that last commit. When rebasing you should ensure that your master branch is up to date with the remote origin branch.

**Comment by nalinbhardwaj** [comment]
Done now @Roasbeef sorry about the weird error, not sure why I got that into commit history in first attempt. It's clean now.

**Comment by Roasbeef** [comment]
Is your PGP key available on any public key server?

**Comment by nalinbhardwaj** [comment]
@roasbeef I have it on [keybase](http://keybase.io/nibnalin)

**Pull request merged**